airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_stream_reader.py

```diff
@@ -54,11 +54,22 @@ def filter_files_by_globs(files: List[RemoteFile], globs: List[str]) -> List[Rem
         """
         Utility method for filtering files based on globs.
         """
-        return [file for file in files if any(fnmatch(file.uri, glob) for glob in globs)]
+        if not globs:
+            return files
```

Comment:
why is this defaulting to all files? I'd expect no globs to mean no files, or to result in a configuration error